### PR TITLE
Added Guest Service Interface control for xVMHyperV resource

### PR DESCRIPTION
**Pull request by bgelens**

### CODE REVIEW

**Comment by bgelens** [comment]
Not sure why Test-xDscSchema and Test-xDscResource threw exceptions. I cannot repro this locally and both return true for me.

**Comment by TravisEz13** [comment]
@bgelens Most likely your schema files are not ANSI/ASCII.  I updated the xDSCResourceDesigner yesterday (1/19) to be more clear about what needs to be done (in the dev branch.)  UTF-16, and ANSI are supported.  For GIT, ANSI is the the best option as it will not be treated as a binary file.

File: README.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ Please see the Examples section for more details.
 
 * xHyperV: Added SecureBoot parameter to enable control of the secure boot BIOS setting on generation 2 VMs.
 -  Fixed drive letter when mounting VHD when calling resource xVhdFile. Fixes #20.
+* xVMHyperV: Added enabling of Guest Service Interface.
 
 ### 3.2.0.0
```
Comment (TravisEz13) [inline README.md]: Add an  section for an 'unreleased' version and summarize the changes for that version.

File: Tests/MSFT_xVMHyper-V/xVMHyper-V.Tests.ps1 [hunk 2]
```diff
@@ -64,6 +67,7 @@ Describe 'xVMHyper-V' {
         Mock -CommandName Get-VM -ParameterFilter { $Name -eq 'NonexistentVM' } -MockWith { Write-Error 'VM not found.'; }
         Mock -CommandName Get-VM -ParameterFilter { $Name -eq 'DuplicateVM' } -MockWith { return @([PSCustomObject] $stubVM, [PSCustomObject] $stubVM); }
```
Comment (TravisEz13) [inline Tests/MSFT_xVMHyper-V/xVMHyper-V.Tests.ps1]: We should have a test to verify that get-targetresource accurately returns EnableGuestService

File: Tests/MSFT_xVMHyper-V/xVMHyper-V.Tests.ps1 [hunk 3]
```diff
@@ -29,6 +29,9 @@ Describe 'xVMHyper-V' {
         function Set-VMNetworkAdapter { }
         function Get-VMFirmware { }
```
Comment (TravisEz13) [inline Tests/MSFT_xVMHyper-V/xVMHyper-V.Tests.ps1]: There should be a tests that verify that test-target resource passes/fails appropriately based on the state of EnableGuestService

**Comment by bgelens** [comment]
Merged with newer commit, Added additional tests as requested and fixed encoding.

File: README.md [hunk 4]
```diff
@@ -78,6 +78,7 @@ Please see the Examples section for more details.
 -  Fixed drive letter when mounting VHD when calling resource xVhdFile. Fixes #20.
 * MSFT_xVMHyperV: Changed the SwitchName parameter to string[] to support assigning multiple NICs to virtual machines.
 * MSFT_xVMHyperV: Changed the MACAddress parameter to string[] to support assigning multiple MAC addresses to virtual machines.
+* MSFT_xVMHyperV: Added enabling of Guest Service Interface.
 
 ### 3.2.0.0
```
Comment (TravisEz13) [inline README.md]: @bgelens Didn't notice this before I merged, 
Please include a summary of changes in an unreleased version change per the [contribution guidelines](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#lifecycle-of-a-pull-request)

**Comment by TravisEz13** [comment]
Thanks for the contribution

**Pull request merged**